src/main/java/Main.java

```diff
@@ -1,205 +1,151 @@
-        System.out.println("continuationToken2: " + continuationToken);
-        map.put(continuationToken, results);
-        return map;
     }
 
-
-
     private void executeSimpleQueryWithList() throws JSONException {
-
-        FeedOptions queryOptions = new FeedOptions();
-        // note that setMaxItemCount sets the number of items to return in a single page result
-        queryOptions.setMaxItemCount(500);
-        queryOptions.setEnableCrossPartitionQuery(true);
-
-        HashMap<Integer, List<JSONObject>> docsPerPage = new HashMap<>();
+        CosmosQueryRequestOptions queryOptions = new CosmosQueryRequestOptions();
 
         String collectionName = "ListItemDetailsCollection";
+        String containerSql = String.format("SELECT * from c where c.id = '%s'", collectionName);
+        System.out.println(containerSql);
+        String itemSql = String.format("SELECT * FROM %s", collectionName);
+        System.out.println(itemSql);
 
-        String collectionLink = String.format("/dbs/%s/colls/%s", databaseName, collectionName);
-        Observable<FeedResponse<Document>> queryObservable =
-                client.queryDocuments(collectionLink,
-                        "SELECT * FROM ListItemDetailsCollection", queryOptions);
-
-        Iterator<FeedResponse<Document>> it = queryObservable.toBlocking().getIterator();
+        CosmosPagedIterable<CosmosContainerProperties> results = database.queryContainers(containerSql, queryOptions);
 
         //1. Create a cache manager
         CacheManager cm = CacheManager.getInstance();
 
         //2. Get a cache called "listDocCache" defined in config
         Cache cache = cm.getCache("listDocCache");
 
-
-        FeedResponse<Document> page = it.next();
-        List<Document> results = page.getResults();
-        System.out.println("count " + results.size());
-        for (Document doc : results) {
-            JSONObject obj = new JSONObject(doc.toJson());
-
-            JSONArray ja_data = obj.getJSONArray("listItemDetails");
-            System.out.println("JSONArray : " + ja_data);
-
-            int length = ja_data.length();
-
-            for (int i = 0; i < length; i++) {
-                JSONObject jsonObj = ja_data.getJSONObject(i);
-                //4. Put  elements in cache
-                cache.put(new Element(i,jsonObj));
-
-                System.out.println("array item: " + jsonObj.toString());
-
-            }
+        // Iterating through all the results.
+        // If there are more than a single page of results, it will auto-magically fetch the next page.
+        for (CosmosContainerProperties containerProperties : results) {
+            String id = containerProperties.getId();
+            CosmosContainer container = database.getContainer(id);
+            container.queryItems(itemSql, queryOptions, JsonNode.class).forEach(item->{
+                try {
+                    JSONObject obj = new JSONObject(item.toString());
```

Comment:
What objects are you trying to deserialize here? How do people set up this this sample? I looked through this repo for a way to upload initial data, but could not find any. 
`queryItems` is capable of deserializing the item into a strongly typed object instead of just an abstract JsonNode.

Reply:
It's similar to the .NET sample where it can deserialize into a strongly typed SalesOrder object, we should have something like this.
https://github.com/Azure/azure-cosmos-dotnet-v2/blob/master/samples/code-samples/DocumentManagement/Program.cs#L218